urc_tf/src/world_frame.cpp

```diff
@@ -0,0 +1,96 @@
+#include "world_frame.hpp"
+
+namespace world_frame {
+WorldFrameBroadcaster::WorldFrameBroadcaster(const rclcpp::NodeOptions &options)
+    : Node("world_frame", options) {
+  tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
+
+  gps_subscriber = create_subscription<sensor_msgs::msg::NavSatFix>(
+      "/fix", rclcpp::SystemDefaultsQoS(),
+      [this](const sensor_msgs::msg::NavSatFix msg) { GPSCallback(msg); });
+
+  waypoint_subscriber = create_subscription<urc_msgs::msg::Waypoint>(
+      "/waypoint", rclcpp::SystemDefaultsQoS(),
+      [this](const urc_msgs::msg::Waypoint msg) { WaypointCallback(msg); });
+
+  set_base_subscriber = create_subscription<std_msgs::msg::Bool>(
```

Comment:
Ideally, this shouldn't be subscribing to a topic. Instead, you'd want to expose a service that sets the base coordinates when called. Maybe something like this?
```
float64 latitude
float64 longitude
float64 altitude
---
std_msgs/Empty response
```